### DISCUSSION TITLE
Native Home Assistant API integration

### DISCUSSION THREAD

**Discussion by lwestenberg** [Hypfer/Valetudo#692] [Archive]
Since the beginning Valetudo connects to Home Assistant with MQTT and it works perfect. Personally I discovered that Valetudo is the only thing still connecting to my HA instance over MQTT. All others are either directly connected with Hassio addons or connecting with the API. 

Perhaps it’s time for a direct integration with HA over the API? That would make running a MQTT broker optional.

**Comment by schinken** [comment]
Thanks for your input, but I also wanted to state my opinion on that issue. It might be, that in your setup valetudo is the only thing that's using a MQTT broker. But I'm pretty sure that in a typical "Smart Home"-Setup with different components, you can't run without a MQTT broker. Also the memory footprint is so low - so it's more like a no-brainer running this.

I understand your problem thou, that you want minimal components in your setup. But from the developer perspective of valetudo, the complexity rises. You have to maintain and test mqtt (ok, thats already implemented, but I wanted to put it on the list ;)). Then you have to maintain code for the hassio-component and a python-written addon in hass.

It raises the complexity for a already good-working solution. Even with autodiscovery etc...  :)

**Reply by lwestenberg** [reply]
Hi, thanks for your reply. I agree that MQTT is popular but not that you can't run without. Most integrations run over local API's or are locally connected (like a Z-Wave or Zigbee stick) and they all communicate via the API. Other devices like ESP's can run on ESPHome which uses the Home Assistant API directly. The advantage is that Valetudo could have it's own HA integration which makes setup more user friendly and the auto discovery for that is also quite simple to create. 

Another idea might be to create a Valetudo HA integration that uses local polling and the Valetudo API. That way you can even save in code maintenance and Valetudo code size because the MQTT part can be real 'clean' MQTT without any Home Assistant specifics like Auto Discovery and Home Assistant topic/message formatting.

I can help creating the integration by the way.
